### ISSUE TITLE
Incorrect type assignability between {} and object

### ISSUE THREAD

**Issue by HugeLetters** [microsoft/TypeScript#56520]
### 🔎 Search Terms

type assignability, {}, object, extends 

### 🕗 Version & Regression Information

This is the behavior in every version I tried, and I reviewed the FAQ for entries about {} and object types

### ⏯ Playground Link

https://www.typescriptlang.org/play?ts=5.3.2#code/C4TwDgpgBAggvAbwL5QgD2BAdgEwM5QD2ARgFYQDGwA-MAE4CuEAXAGYCGANnhAFAD0-KMIB61XkA

### 💻 Code

```ts
type A={} extends object?true:false
```

### 🙁 Actual behavior

type A is resolved to true even though type `{}` is a broader type than `object`. Strings are `{}` but they aren't `object` for example. 

### 🙂 Expected behavior

type A should be resolved as false. 
Otherwise it leads to this bug for example
```ts
function hasProp(obj: {}, key: string) {
	return key in obj;
}

hasProp('string', 'key');
```

Typescript doesn't report any errors here but it will cause a runtime error.

**Comment by fatcerberus** [comment]
By design, despite the unsoundness.  Primitive strings are also assignable to `{ toString(): string }`, for example, and it’s much harder to make the case that *that* shouldn’t be assignable to `object`.

**Comment by HugeLetters** [comment]
> By design, despite the unsoundness. Primitive strings are also assignable to `{ toString(): string }`, for example, and it’s much harder to make the case that _that_ shouldn’t be assignable to `object`.

Good point, thank you!

**Comment by MartinJohns** [comment]
Related: #43210
